### ISSUE TITLE
Personal Notifications

### ISSUE THREAD

**Issue by tecnobrat** [integrations/slack#380]
Perhaps this is already possible but I'm just not aware.

I would love if I could receive a private message from the Github bot when I've been notified on one of my organizations repos.

Basically, I don't want to have to watch my email for when someone is @mentioning me on github for work.  I would rather those be rolled up into a private message from the Github bot.

Configuration around this so that I could only see notifications for my organizations' repos or all repos would be the ultimate.

**Comment by ryandeussing** [comment]
You mean personal notifications, not personal repos ;) 

But I agree with you and will go even further: each person on my team needs to be able to opt-in or out of notifications, by reason:

> `assign` | You were assigned to the Issue.
> `author` | You created the thread.
> `comment` | You commented on the thread.
> `invitation` | You accepted an invitation to contribute to the repository.
> `manual` | You subscribed to the thread (via an Issue or Pull Request).
> `mention` | You were specifically @mentioned in the content.
> `state_change` | You changed the thread state (for example, closing an Issue or merging a Pull Request).
> `subscribed` | You're watching the repository.
> `team_mention` | You were on a team that was mentioned.

We currently accomplish this by using Gmail filters and a Slack email address created for each user that posts to a different room for each user. It's hacky.

**Comment by pavanvamsi3** [comment]
Is @mention available now? Can I enable it?

**Comment by Hollywood** [comment]
Gentle bump on this. Internally this would help the `services` team tremendously in regards to keeping up with comms and allowing us to respond within a reasonable timeframe!

**Comment by steve-sweriduk-zocdoc** [comment]
Agreed. Would be really nice to have either a private channel or just slackbot notify my when somebody comments on my PR or the code, requests a review, merges or closes it. Right now, the only workaround I can think of is configuring Zapier to look at my email and notify me from there.

**Comment by sklarsa** [comment]
Any progress on this? Is there any way I can help?  This would be huge for us in filtering out noise in github activity streams

**Comment by waldyrious** [comment]
@wilhelmklopp, @bkeepers -- any chance you could give us an idea of what would be necessary for this to happen? What can we do to help move this forward?

**Comment by allbombs** [comment]
Subscribed, want this!!

**Comment by daveknapik** [comment]
Subscribed. Please make it happen! Thanks!! :)

**Comment by abinoda** [comment]
Just an FYI that [Pull Reminders in the GitHub Marketplace](https://github.com/marketplace/pull-reminders/) does this.

You can subscribe to DM notifications for almost any pull request related event (see below) including comment mentions, replies, merge conflicts, and failed CI checks.

I maintain Pull Reminders so if pricing is problematic for anyone, please email me at abi@pullreminders.com and I'd be happy to give your team a free account.

<img width="825" alt="screen shot 2019-01-08 at 1 29 43 am" src="https://user-images.githubusercontent.com/50083/50816088-1889b700-12e5-11e9-82d7-39448df1911d.png">

**Comment by Hollywood** [comment]
@abinoda just wanted to say I loved your talk at Universe.

**Comment by abinoda** [comment]
@Hollywood Thanks! Btw, if you want to use Pull Reminders internally at GitHub you're more than welcome to free licenses abound. It works on-prem too: https://github.com/pullreminders/enterprise

**Comment by mkoleva** [comment]
We have started using PullReminders and it satisfies all our needs - gives us the personal notifications we wanted and the stats help us identify bottlenecks. It's super convenient. Thanks @abinoda !

**Comment by Vict0rSch** [comment]
Unfortunately, though Pull Reminders works for PRs, issues are still not customizable as requested here

**Comment by abinoda** [comment]
@vict0rsch Don't want to sidetrack this issue thread but did want to respond to say that we expect personal notifications for Issues to be added in the very near-term.

**Comment by melnychukvitaliy** [comment]
Did you plan work on it?
Want this feature, is there any way I can help?
I would like to join

**Comment by leifhanack** [comment]
FYI - The company behind Pull Reminders is joining GitHub. The product is free now and its features will be integrated into GH core: https://pullpanda.com/github

**Comment by jrochkind** [comment]
@leifhanack I coincidentally just arrived at this issue looking for this func, two hours after you posted that. 

Great!

Note for others, "Pull Panda" really is the same github marketplace app as "Pull Reminders" mentioned above. The above link to a "Pull Reminders" github marketplace app actually takes you to one called "Pull Panda", I'd guess it got renamed sometime after the earlier comment was made.

**Comment by scepticulous** [comment]
Hi everybody 👋 ,

thanks for all your great input.
It is correct, that Pull-Panda joined GitHub and Pull-Reminders are part of Pull-Panda.
The different parts are going to be integrated into GitHub, but this does not mean, that there is no room for personal notifications in the Slack integration.

Some kind of personal notifications for certain events is on the roadmap of things we want to improve. I don't have an ETA right now, but I will keep y'all posted.

**Comment by dimitribouniol** [comment]
For what it's worth, you can subscribe within the private GitHub app-channel, and only you will get the notifications you subscribe to.

**Comment by hbd** [comment]
Any update on this? Now that Pull Reminders is not available in the Marketplace, there doesn't seem the be a good option for personal notifications.

**Comment by MichaelViveros** [comment]
I also would like an update. My company installed Pull Reminders in October but now new people that join my company can't sign up for Pull Reminders and there's no way for them to setup similar direct message notifications using the Github Slack bot so it's really killing their productivity.

**Comment by Hollywood** [comment]
@hbd we're currently working on integrating it into [GitHub](https://pullpanda.com/github). Checkout [Pull Panda](https://pullpanda.com) in the meantime. It's Pull Reminders but evolved into more and the best part is that it's **free**!

**Comment by andnor** [comment]
@Hollywood 

[Pull Panda](https://pullpanda.com/) is not accepting signups 😞 

Is there another way to use the product? 
Does it support notifications on 'issues' yet?

**Comment by glaebhoerl** [comment]
Just googled for this feature and found this issue. I would also really appreciate having this functionality available.

**Comment by pytzcarraldo** [comment]
I guess I'll give this another bump - would be a really awesome feature to have available!

**Comment by vict0rsch** [comment]
How is this still not a thing 2+ years after issue was opened and clear demand from the community?

**Comment by SavePointSam** [comment]
While waiting for this feature, I have found [this project](https://github.com/manosim/gitify) to be very useful. It doesn't send your notifications to slack, rather integrates with the MacOS notification system which works well enough for me given this feature doesn't exist.

**Comment by jean-sandberg** [comment]
> For what it's worth, you can subscribe within the private GitHub app-channel, and only you will get the notifications you subscribe to.

Yes, but I am unable to subscribe only to comments where I am @ mentioned. Can't wait for this feature! Yesterday I missed a GH email where I was @ mentioned and it delayed someone's work. If I were able to get a Slack notification I could have responded right away. Thanks!!

**Comment by mressler** [comment]
My team switched to GitHub recently. Really glad we did with remote work really taking off. The only downside and complaint from my team in retrospectives is exactly this. If we could just get an @mention in slack when we get an @mention in GitHub, our lives would be complete.

Bump and context. Thanks to anyone already working on this feature!

**Comment by EtienneM** [comment]
This is now possible by enabling this setting: https://github.com/settings/reminders

The only thing I wish they would implement is that they send the reminder daily, but I would also love to be able to ask for the content of the reminder whenever I want, for instance with a command like `/github reminder`.

Otherwise it works great! :)

**Comment by glaebhoerl** [comment]
@EtienneM That URL is a 404 for me?

**Comment by EtienneM** [comment]
> @EtienneM That URL is a 404 for me?

Oh maybe they are doing rolling release of this feature or it's a beta. I don't know :man_shrugging:

**Comment by jackwilsdon** [comment]
[It's a beta but you should be able to sign up for it](https://github.blog/changelog/2019-11-14-scheduled-reminders-beta/)

**Comment by glaebhoerl** [comment]
I'm looking at [the docs](https://help.github.com/en/github/setting-up-and-managing-organizations-and-teams/managing-scheduled-reminders-for-pull-requests) and it appears that this functionality is about sending reminders on a schedule for pending reviews.

Whereas the functionality I'm interested in would be immediately sending a message when @-mentioned. (Ideally also, in a configurable way, when a review gets assigned to you, when someone replies to you, and so on. But @-mentions are the core of it.)

**Comment by jackwilsdon** [comment]
Ah yes you're right @glaebhoerl - this only supports reminding you about review requests.

**Comment by waldyrious** [comment]
@glaebhoerl @jackwilsdon hmm, I seem to have this option under the Real-time alerts feature:

![Screenshot 2020-04-06 at 12 19 15](https://user-images.githubusercontent.com/478237/78553482-38b66780-7801-11ea-8839-3959147648ef.png)

Am I missing something?

**Comment by jackwilsdon** [comment]
Oh wow you're right - I completely missed the "Enable real-time alerts" checkbox 🤦‍♂

**Comment by glaebhoerl** [comment]
Oh, nice! Yeah that looks like what I wanted. :) Apparently they just haven't written the docs for that yet (or at least neither Ctrl-F nor GH's own search finds it). I'll see if I can give it a try!

**Comment by glaebhoerl** [comment]
No dice unfortunately. The organization admin successfully signed up for the beta, but when we go to Settings -> Scheduled reminders in our individual profiles, it says "Upgraded needed" in red text next to the organization name, with the Edit button disabled, and no other information. We have a Team account, which is supposed to be sufficient according to [the docs](https://help.github.com/en/github/setting-up-and-managing-organizations-and-teams/managing-scheduled-reminders-for-pull-requests).

**Comment by jackwilsdon** [comment]
I'd try contacting support in that case - I've managed to enable it for one of my organisations (it's a free one though).

**Comment by ethanjyx** [comment]
yes the scheduled reminders work extremely well, this issue could be closed now

**Comment by michaelfarrell76** [comment]
I previously used pull pandas and have recently switched over to the native support. One minor regression ive noticed is that notifications for PR comments link generally to the pull request, not the specific comment like they did on pull pandas.

**Comment by vict0rsch** [comment]
@ethanjyx before closing, does it work for issues though? like when you're assigned an issue or mentioned in one?

**Comment by ethanjyx** [comment]
@vict0rsch no it doesn't

**Comment by ashishkeshan** [comment]
Would also love to see an integration for team mentions on issues!

**Comment by virtualguy** [comment]
Looks like "Realtime Alerts" are gone, i'm really hoping this is temporary as it was a killer feature!

**Comment by timharris777** [comment]
I confirm the realtime DM alerts are not in Github's native pull panda features.

**Comment by nasht00** [comment]
I am interested in this, for GitHub Enterprise.
Is this feasible right now?

**Comment by RayBB** [comment]
Does anyone know if there's an easy way to get notifications from things you're watching that aren't in an org?

When I go to https://github.com/settings/reminders I only see settings for github orgs I'm a part of.

I'd really like to get notifications when someone mentions me on public repos where I'm not part of the github org the repo is in.

**Comment by ntwong0** [comment]
> Does anyone know if there's an easy way to get notifications from things you're watching that aren't in an org?
> 
> When I go to https://github.com/settings/reminders I only see settings for github orgs I'm a part of.
> 
> I'd really like to get notifications when someone mentions me on public repos when I'm not part of the github org the repo is in.

I wonder if there's a way to extend the `comments` feature to filter only for @mentions

**Comment by littlehome-eugene** [comment]
Why am I not getting slack notification

I can see that slack workspace is connected .
But no channel or my id is not specified, how does github notify me through slack when it doesn't know which channel it's going to use to talk to me?

<img width="1310" alt="Screen Shot 2020-08-16 at 2 43 33 PM" src="https://user-images.githubusercontent.com/24217396/90327497-f54de000-dfce-11ea-8a35-0663244b02de.png">

**Comment by littlehome-eugene** [comment]
comments sends slack notification when comments are made on  pull requests not just any issues? 

I don't understand why I'm not getting slack notification whereas people here seem to be enjoying github-slack notification integration.

**Comment by joeyfezster** [comment]
I just set this up just like @littlehome-eugene stated, and tested it by creating a comment with a @mention in a file in a random commit (for example [here](https://github.com/google/googletest/commit/465ce90c92979bc0015c1d3354109cbffbc8bc3b)). Slack does not notify.

**Comment by joeyfezster** [comment]
For anyone on the team:

As a remote team, we rely on comments left on random commits with mentions just like [this one](https://github.com/google/googletest/commit/465ce90c92979bc0015c1d3354109cbffbc8bc3b#r41644628).
This is super important for our workflow, and yet the functionality is buggy/flaky/incomplete.

A comment with a mention like that one is virtually invisible except if someone sends you a link to it. Not the GHMobile app nor the integration mention above notify my for these comments, despite both being properly configued

Please help

**Comment by kc1116** [comment]
Can you enable real time alerts on an organization level?

**Comment by karfau** [comment]
It's also possible that @kc1116 is referring to the following scenario (which is a paintpoint we also have):
We want that every dev in our team or org receives personal notifications from the github integration (somehow it seems to be possible via personal scheduled reminders, don't ask for details yet, when I figure them out I'll update here), but we want to not wait until every developer has configured that on her/his own but want to configure it for everybody, ideally even for everybody that is joining the team/org.

**Comment by ahill187** [comment]
> Why am I not getting slack notification
> 
> I can see that slack workspace is connected .
> But no channel or my id is not specified, how does github notify me through slack when it doesn't know which channel it's going to use to talk to me?
> 
> <img alt="Screen Shot 2020-08-16 at 2 43 33 PM" width="1310" src="https://user-images.githubusercontent.com/24217396/90327497-f54de000-dfce-11ea-8a35-0663244b02de.png">

This is a good start, would it be possible also to add an option to get the status of your open pull requests? Under `Review requests assigned to you`, a third option such as `Your open pull requests`.

Also, I like the scheduling, but the old `Pull Reminders` had a command in `Slack` which allowed you to list any of these types of notifications. For example, you could type `/some-command` and it would list all your open pull requests ad hoc.

**Comment by 0xdevalias** [comment]
It seems that the https://github.com/settings/reminders feature only works for public repositories, unless you upgrade to a paid plan. While I do understand the desire for the upsell, as a mostly open-source project that happens to track our backlog in a private repo, this means that most of the times where realtime `@mention` support would be useful, isn't possible for us.

![image](https://user-images.githubusercontent.com/753891/110048460-0102c780-7da4-11eb-9602-d5c4d010e92e.png)

If we were able to enable it via this integration, then we could have the best of both worlds. So would still be a super useful feature to get enabled.

**Comment by Kishan1311** [comment]
![image](https://user-images.githubusercontent.com/48950421/116009093-7a35d280-a60f-11eb-8ae1-ee60712cfd76.png)

Hi all, what should I write in this text field to get notified about failed CI checks. I tried: `ci/circleci: compatibility-tests` but it doesn't seem to work. Thanks in advance

**Comment by Habitats** [comment]
Is this the right thread to follow the progress on getting notified on @mentions in _issues_ and not just @mentions in PRs?

It's extremely frustrating to not get notifications on issues as well, so we're really looking forward to that one!

**Comment by Nogias9x** [comment]
Please make "personalized notifications" happen :pray: :pray: :pray:

**Comment by Obsecurus** [comment]
For anyone else landing here I just found Toast - https://toast.ninja/ and it fixes this problem although $4/mo per user after 3 users on private repos.  I'm guessing Microsoft will acquire them like they did with pullpanda.

**Comment by acoudouy** [comment]
Hey for anyone looking for better notifications on pull requests and working on Slack, we recently launched our beta on Axolo. The beta is free. 

Axolo creates temporary Slack channels, invite only people that should be, and sends all Github notifications in their dedicated channels. You can select/unselect which kind of Github Action or Deployment you want to receive too (success, error, pending, ...). I'm looking for feedback so feel free to contact me! Thanks

**Comment by fredrikaverpil** [comment]
I just found realtime, personal notifications under my org. Is this new?
https://github.com/settings/reminders/ (click the edit/pen icon)

![image](https://user-images.githubusercontent.com/994357/145409957-440b4fb5-65ef-4574-86ea-8a34cf066ad5.png)

This is everything I ever wanted.

**Comment by Habitats** [comment]
@fredrikaverpil problem is that doesn't work for _mentions_ on _issues_ -- which is the entire problem :\

**Comment by fredrikaverpil** [comment]
Ah, darn... right. One can only hope the only reason we are not seeing them in the list there is because of GitHub still rolling out the [new issues system](https://github.blog/2021-06-23-introducing-new-github-issues/).

**Comment by kostecky** [comment]
Please make personalized notifications happen! Specifically @mentions going to slack DMs!

**Comment by kc1116** [comment]
> It's also possible that @kc1116 is referring to the following scenario (which is a paintpoint we also have): We want that every dev in our team or org receives personal notifications from the github integration (somehow it seems to be possible via personal scheduled reminders, don't ask for details yet, when I figure them out I'll update here), but we want to not wait until every developer has configured that on her/his own but want to configure it for everybody, ideally even for everybody that is joining the team/org.

This is exactly what I was referring to.

**Comment by emekdahl-palmetto** [comment]
bump on at mentions for slack subscriptions

**Comment by lmossman** [comment]
Bumping this - slack notifications for @mentions on github **issues** is greatly desired. Please let us know if this is being worked on!

**Comment by tedspare** [comment]
Notifications for @mentions on issues work in [Neat.run](https://neat.run) on macOS:

<img src="https://user-images.githubusercontent.com/36117635/150439441-b2a364ac-8d4e-45b4-a036-0ac3d623d189.png" width="550"/>

Self-plug but it seems a great number of us experience this!

**Comment by angalic** [comment]
Bump @mentions on github issues.

**Comment by gazreyn** [comment]
+1 on this one. Super useful when we have team members who don't keep as on top of emails as they should but ever since Slack Notifications has never been late for a meeting. Would love to be able to send DM's when mentioned on an issue comment.

**Comment by shridharpuntambekar-okta** [comment]
+1 for this issue.
I am surprised this request has been open for 4 years now especially when the company providing the specific functionality was merged with the company!
This functionality should be available by default and is a MUST to sort through the clutter of "Scheduled Notification".

**Comment by aasiddiq** [comment]
Hi folks
From what I understand this request was for supporting @ mentions in issues and comments. This feature was added few months back, you can read more about it in [Readme](https://github.com/integrations/slack#mentions).
If you are facing any issue in using the feature feel free to reopen this.

**Comment by wmertens** [comment]
@aasiddiq how is the mentioning supposed to work? I get the correct @ in slack when I'm added as a reviewer, but I don't get a notification when someone mentions me in a comment.
Am I supposed to do some personal subscribe somewhere?

**Comment by aasiddiq** [comment]
@wmertens to get comment notification you need to subscribe to the `comments` feature as well.
You can check the current subscription in the channel using 

`/github subscribe list features`

and subscribe to comments using

`/github subscribe <org/repo> comments`

**Comment by wmertens** [comment]
@aasiddiq but that will put all comments in Slack right?

Am I understanding correctly that there's no way to get a private notification when you're mentioned?

**Comment by gazreyn** [comment]
Just following up with what @wmertens said. From what I can see, this will subscribe to all comments in the repo. Is there no way to subscribe only to comments where you're mentioned?

**Comment by aasiddiq** [comment]
Unfortunately, no that is not an option right now.

**Comment by NathanHazout** [comment]
So this is not relevant to this thread which is about "personal notifications"?

**Comment by Habitats** [comment]
Please reopen this issue because it is obviously not solved, and clearly something a lot of the community is requesting.

**Comment by daniellapZ** [comment]
Bump on @mentions at the slake channel as part of the scheduled reminders! 🙆‍♀️
Will be a great help to track the discussions taking place without checking constantly your mail.

Is there any update where this effort stands? an ETA?

**Comment by smartinio** [comment]
@aasiddiq Any ETA on when mentions on issues (like I just did) will be implemented? I can sign an NDA and help build the integration for free if you don't have enough resources.

**Comment by aasiddiq** [comment]
Tagging @ashokirla for checking on the ask here

**Comment by 0xdevalias** [comment]
> I would love to get a notification every time I am mentioned _anywhere_ on GitHub, essentially integrating this: [github.com/notifications?query=reason%3Amention](https://github.com/notifications?query=reason%3Amention) with a private DM on Slack

This is basically what I would love to be able to do as well.

---

Here's the relevant GitHub API docs related to notifications:

- https://docs.github.com/en/rest/activity/notifications

We can list notifications for the current user, and could filter them with the `since` param:

- https://docs.github.com/en/rest/activity/notifications?apiVersion=2022-11-28#list-notifications-for-the-authenticated-user

It's also possible to scope this down to only be in terms of a specific repository:

- https://docs.github.com/en/rest/activity/notifications?apiVersion=2022-11-28#list-repository-notifications-for-the-authenticated-user

---

Using GitHub actions, these are the following ways that we can trigger an action to run:

- https://docs.github.com/en/actions/using-workflows/triggering-a-workflow
  - > Events that occur in your workflow's repository
  - > Events that occur outside of GitHub and trigger a `repository_dispatch` event on GitHub
  - > Scheduled times
  - > Manual
- https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows

While it doesn't look like any of the event types we can use would allow us to trigger the GitHub action directly when a notification is received, we could poll the notifications endpoint using the `schedule` event (this would likely be best used as a fallback for cases not covered by the other more specific events):

- https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#schedule

There are also more specific event types that would let let our GitHub action be triggered sooner:

- https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#discussion_comment
- https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#issue_comment
- https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_comment-use-issue_comment
- https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_review
- https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_review_comment

**Comment by tedspare** [comment]
@0xdevalias you're describing the architecture of [Neat](https://neat.run) 😅 (cc @apolinario)

**Comment by 0xdevalias** [comment]
> you're describing the architecture of [Neat](https://neat.run)

@tedspare Ha, interesting.. Shame they've basically abandoned the project now though (with hopes/plans to open-source it 'at some point'). From the community slack:

> Hey there @channel! We have a much-belated update
> Since starting 17 mo ago, over 2.2k engineers like you have now used Neat. As a team of four new grads, we opted to take the venture route, living on a $200k pre-seed round. And then we ran out.
> Now, to keep the Neat business running, we’ve pivoted to an agency model: [studio.neat.run](https://studio.neat.run/). This means our support of Neat the product will be minimized, and this Slack will wind down.
> We now believe more than ever that a more sustainable Neat 2.0 will be open-ended, open-source, and community-backed. In the meantime, we’ll prioritize bugfixes.
> Above all, we want to thank you for supporting us. Your input has impacted hundreds of others. If you liked Neat and want to work together, we’re open to building new products. I’ll be at [ted@neat.run](mailto:ted@neat.run). :)

**Comment by smartinio** [comment]
> Tagging @ashokirla for checking on the ask here

@aasiddiq @ashokirla Hello again. Is there any plan for integrating this? Again, if there aren't enough resources, there are people here who would be able to help build this.
